### ISSUE TITLE
Plot recorded with recordPlot() not showing in Rnotebook

### ISSUE THREAD

**Issue by brendaprallon** [rstudio/rstudio#5648]
I have previously asked this question on [Stack Overflow](https://stackoverflow.com/questions/58175496/plot-recorded-with-recordplot-not-showing-in-rnotebook) and on [RStudio Community](https://community.rstudio.com/t/plot-recorded-with-recordplot-not-showing-in-rnotebook/41211). It's been 8 days and no answers, so here I am :)

I am having trouble displaying plots inline on R notebook that I have created using the recordPlot() function. The chunks run without error but the recorded plot does not show. Below is a toy example. The second chunk does not output the plot inline, as I would expect. But if I run the line on the console, the plot is displayed on the plot window.
````
---
title: "R Notebook"
output: html_notebook
---

```{r, echo=FALSE}
plt_list <- list()
for (i in 1:3){
    plot(c(i, i+1, i+2))
    plt_list[[i]] <- recordPlot() 
}
```

```{r}
plt_list[[1]]
```
````
I have, since I asked the question on RStudio Community, updaded my RStudio and packages, and I have also tried the development version of rmarkdown as suggested here. The problem still persists. 

```
xfun::session_info('rmarkdown')
R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.3 LTS

Locale:
  LC_CTYPE=pt_BR.UTF-8       LC_NUMERIC=C               LC_TIME=pt_BR.UTF-8       
  LC_COLLATE=en_US.UTF-8     LC_MONETARY=pt_BR.UTF-8    LC_MESSAGES=en_US.UTF-8   
  LC_PAPER=pt_BR.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
  LC_TELEPHONE=C             LC_MEASUREMENT=pt_BR.UTF-8 LC_IDENTIFICATION=C       

Package version:
  base64enc_0.1.3  digest_0.6.21    evaluate_0.14    glue_1.3.1       graphics_3.6.1  
  grDevices_3.6.1  highr_0.8        htmltools_0.4.0  jsonlite_1.6     knitr_1.25      
  magrittr_1.5     markdown_1.1     methods_3.6.1    mime_0.7         Rcpp_1.0.2      
  rlang_0.4.0      rmarkdown_1.16.1 stats_3.6.1      stringi_1.4.3    stringr_1.4.0   
  tinytex_0.16     tools_3.6.1      utils_3.6.1      xfun_0.10        yaml_2.2.0      


Pandoc version: 2.3.1`
---
 ```

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.name/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('rmarkdown')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/rmarkdown')`.
   - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by dfalty** [comment]
I can reproduce this on Mac 1.2.5018.

**Comment by cderv** [comment]
This is still happening in 1.4.1554

**Comment by cderv** [comment]
Still an issue that may be worth looking into.

**Comment by kevinushey** [comment]
Does it reproduce in the latest development versions of R? There was a recent change to how `replayPlot()` initializes a new graphics device that may be relevant here...

**Comment by cderv** [comment]
Good call! I do not follow closely what happens in r-devel. 

Using the last build of r-devel, the recorded plot will show in the plot pane in the IDE now when the code is executed in the console. 

However, this is still an issue inside the `html_notebook` format in source pane. Running the chunk with the recorded plot shows nothing in Notebook mode

**Comment by kevinushey** [comment]
If I understand correctly, the following should be a simpler reproducible example:

`````
---
title: "R Notebook"
output: html_notebook
---

```{r}
plot(1)
plt <- recordPlot()
```

```{r}
replayPlot(plt)
```
`````

Calling `replayPlot()` from the console produces a plot in the Plots pane, but seems to do nothing within a notebook chunk. My guess is that this manipulates the graphics device in a way that our Notebook pseudo-graphics-device fails to see.

**Comment by cderv** [comment]
Yes this is a simpler reproducible example, with calling `replayPlot` directly. Print method `grDevices:::print.recordedplot()` calls `replayPlot()`

Thanks @kevinushey !

**Comment by danieltomasz** [comment]
Any progress in fixing it?

**Comment by danieltomasz** [comment]
A  bit cumbersome workaround (not solving problem, but enabling the preview)
```{r}
plot(1)
plt <- recordPlot()
```
And view  in another cell
```{r}
grDevices::png("plot.png", width = 600, height = 400)
replayPlot(plt)
dev.off()
knitr::include_graphics("plot.png")
```

**Comment by avishai987** [comment]
> A bit cumbersome workaround (not solving problem, but enabling the preview)
> 
> ```
> plot(1)
> plt <- recordPlot()
> ```
> 
> And view in another cell
> 
> ```
> replayBasePlot <- function(plt) {
>   tmp_file <- tempfile(fileext = ".png")
>   grDevices::png(tmp_file, width = 600, height = 400)
>   replayPlot(plt)
>   dev.off()
>   knitr::include_graphics(tmp_file)
> }
> 
> replayBasePlot(plt)
> ```

Also with 
```
plot.new()
recorded_plot
```
though it produces an extra blank plot...
